src/TableGenerator.cpp

```diff
@@ -1,24 +1,37 @@
 #include "TableGenerator.h"
 #include "random.h"
 
-Table TableGenerator::generateTable(unsigned int numOfLocalColumns, unsigned int numOfRemoteColumns, unsigned long numOfRows, unsigned int maxRandomNumberInCell) {
+Table TableGenerator::generateTable(unsigned int numOfLocalColumns, unsigned int numOfRemoteColumns,
+                                    unsigned long numOfRows, unsigned int maxRandomNumberInCell, int numaNode) {
     srand(time(NULL));
     Table table;
     for (unsigned int i = 0; i < numOfLocalColumns; ++i) {
-        auto column = std::make_shared<Column<uint32_t>>(numOfRows, 0);
-        for (unsigned long j = 0; j < numOfRows; ++j) {
-            uint32_t cellValue = (uint32_t) Random::next() % maxRandomNumberInCell;
-            column.get()->data().at(j) = cellValue;
-        }
-        table.addColumn(column);
+        addColumn(numOfRows, maxRandomNumberInCell, 0, table);
     }
     for (unsigned int i = 0; i < numOfRemoteColumns; ++i) {
-        auto column = std::make_shared<Column<uint32_t>>(numOfRows, numa_max_node());
-        for (unsigned long j = 0; j < numOfRows; ++j) {
-            uint32_t cellValue = (uint32_t) Random::next() % maxRandomNumberInCell;
-            column.get()->data().at(j) = cellValue;
-        }
-        table.addColumn(column);
+        addColumn(numOfRows, maxRandomNumberInCell, numaNode, table);
     }
     return table;
 }
+
+Table TableGenerator::generateTable(unsigned int numOfLocalColumns, unsigned int numOfRemoteColumns,
+                                    unsigned long numOfRows, unsigned int maxRandomNumberInCell) {
+    srand(time(NULL));
+    Table table;
+    for (unsigned int i = 0; i < numOfLocalColumns; ++i) {
+        addColumn(numOfRows, maxRandomNumberInCell, 0, table);
+    }
+    for (unsigned int i = 0; i < numOfRemoteColumns; ++i) {
+        addColumn(numOfRows, maxRandomNumberInCell, (((int) Random::next()) % (numa_max_node() + 1)), table);
```

Comment:
Not quite, it should be `(Random::next() % numa_max_node()) + 1`
The +1 in the `numa_max_node` would give us 0 to 3 and the +1 at the end could give us a 4. So we want the range (0 to 2) and then +1 to give us 1 to 3, which are all valid and not local.